config/dictionaries/resource.json

```diff
@@ -288,14 +313,16 @@
       {
         "id": "move-device",
         "summary": "Move device to another application",
+        "description": "",
         "method": "PATCH",
         "endpoint": "/v5/device(<ID>)",
         "filters": "",
         "data": "{\n    \"belongs_to__application\": \"<APP ID>\"\n}"
       },
       {
         "id": "set-device-to-release",
-        "summary": "Set device to a specific release",
+        "summary": "Pin device to a specific release",
+        "description": "Making a GET request to fetch current target release hash of the device will output as null unless the device is pinned to a release.",
```

Comment:
The summary and the example are for the PATCH method (setting `should_be_running__release` to a given release ID), but the proposed description regards the GET method and the fact that GET will return a `null` value for `should_be_running__release` if this field has not been (explicitly) set. Almost like a description for a different example / different method. :-) But I also see how a user might think that `should_be_running__release` would always have a non-null value (provided that the application has at least one release).

Instead of this description for the PATCH method, how about adding an example for the GET method, immediately before or after the example for the PATCH method? Then the description for the GET could more justifiably refer to the (un)expected results of the GET method. For example:

```
      {
        "id": "get-target-release",
        "summary": "Get the release that a device is pinned to",
        "description": "Note that `should_be_running__release` will be unset (`null`) 
                        if the device has not been explicitly pinned to a release.
                        To find out the target release for an unpinned device, check
                        the `commit` field of the `application` resource instead.",
        "method": "GET",
        "endpoint": "/v5/device(<ID>)",
        "filters": "?\\$select=should_be_running__release",
        "data": ""
      }
```
I believe that _"check the `commit` field of the `application` resource instead"_ is correct, but I am not 100% sure. I have successfully run these `curl` command lines:
```
$ curl "https://api.balena-cloud.com/v5/device(3991606)?\$select=should_be_running__release" \
  -H "Authorization: Bearer $authToken" | jq .
{
  "d": [
    {
      "should_be_running__release": null,
      "__metadata": {}
    }
  ]
}

$ curl "https://api.balena-cloud.com/v5/application(1301645)?\$select=commit" \
  -H "Authorization: Bearer $authToken" | jq .
{
  "d": [
    {
      "commit": "5dadd8fc163e9f5c98f2f417a517193c",
      "__metadata": {}
    }
  ]
}
```

Reply:
Good one, Paulo. Having another example for GET method of the field makes more sense. A caveat I experienced was that we can't format any information inside the description. I tried adding <pre></pre> tags to get preformatted text formating on several terms but it seems like the StaticJinja template directly renders strings out of JSON.

I will add your example, though I recommend not using `commit` field since that's for the entire application, we already have `is_on__commit` for that. So will use that instead (On more thought, I will add an example for that too!)

Reply:
![Screenshot from 2020-10-12 11-45-39](https://user-images.githubusercontent.com/22801822/95711488-7d273200-0c80-11eb-8ddc-305545659f04.png)
This is what I meant by no formatting

Reply:
Let me know if I should create an issue for this or just leave it with these `  quotes

Reply:
> _I recommend not using `commit` field since that's for the entire application, we already have `is_on__commit` for a particular device. So I will use that instead, let me know your thoughts_

Interesting. I have more questions than answers, which is why we need more documentation. :-) Considering the distinction between "current state" and "target state", `is_on__commit` sounds like current state (the commit that the device **is** currently running), whereas `should_be_running__release` sounds like "target state" (the commit† that a device **should** be running). Is it true / correct that `is_on__commit` regards current state, whereas `should_be_running__release` regards target state? For a non-pinned device, I thought that `application.commit` would be the target state for a device that belongs to that application, hence my previous suggestion: _"To find out the target release for an unpinned device, check the `commit` field of the `application` resource instead."_

† Btw, there is also the question of release ID vs commit hash. `should_be_running__release` contains a release ID:
```
$ curl -H "Authorization: Bearer $authToken" -H "Content-Type: text/plain" "https://api.balena-cloud.com/v5/device(3991606)?\$select=should_be_running__release" | jq .
{
  "d": [
    {
      "should_be_running__release": {
        "__id": 1552722,
        "__deferred": {
          "uri": "/resin/release(@id)?@id=1552722"
        }
      },
      "__metadata": {}
    }
  ]
}
```
... whereas `device.is_on__commit` and `application.commit` are commit hashes. But we can probably let users figure this out for themselves... 🤔

Reply:
Yes, you are correct in your assumption @pdcastro, but to not document `should_be_running__release` means we assume the customer knows an intricate bit of how balena works by pinning releases and targeting it and assume that they know why `should_be_running__release` is coming out to be null (because they didn't give a target in the first place) Which brings us all the way back to how the issue came to be #1496 

> For a non-pinned device, I thought that application.commit would be the target state for a device that belongs to that application, hence my previous suggestion: "To find out the target release for an unpinned device, check the commit field of the application resource instead."

You are also right about this, but I still like to go for `is_on__commit` if I want to know what commit the device is running on **right now** 

Good catch on `release ID vs commit hash` and yes, we can let them figure it out this one with minor improvement in the docs to mention it for them. Also, where can I find a device ID? I didn't know they even existed until I made the API call.

Reply:
What about formatting, Paulo? Leave it with single quotes or no single quotes?

Reply:
Regarding formatting, I think the visible `<pre>` elements look bad. Visible backticks like \`should_be_running__release\` look less bad, :-) but probably also better without them. So, short of fixing the system to allow markdown formatting, my opinion is to remove the quotes.

> You are also right about this, but I still like to go for `is_on__commit` if I want to know what commit the device is running on **right now**

hmm Not sure we are on the same page here. :-)  Comparing these two statements:

* _"To find out the **target** release for an unpinned device, check the `commit` field of the `application` resource instead."_
* _"To find out the **target** release for an unpinned device, check the `is_on__commit ` field instead."_

I suspect that one of the statements above is incorrect -- we should clarify which one before advising users. :-) The statements above offer an alternative to using `should_be_running__release` when it is `null` (because the device is unpinned). ***If it is true that `should_be_running__release` regards target state rather than current state,*** then we should advise users how to obtain a device's **target** state (possibly `application.commit`), rather than a device's **current** state (possibly `device.is_on__commit`).

Note that I am assuming, but have not verified, that:

* `device.should_be_running__release` regards target state
* `device.is_on__commit` regards current state
* `application.commit` regards target state

I leave it on you to verify that the above is correct. :-)  If these assumptions are correct, the description field could actually advise users further:

_"For an unpinned device, the `device.is_on__commit` field may be used to obtain the device's current state, while the `application.commit` field represents the device's target state."_

Again, I don't know that the above is true, please verify. :-)

Reply:
> "To find out the target release for an unpinned device, check the is_on__commit field instead."

This is incorrect, in my statement earlier I was showing my preference for `is_on__commit` if I want to find the current state of the device that's all. I agree with everything you mentioned above including recommending users an alternative if one is not available. I am making a final change, after which I can ping on beginners for folks to verify it?

I have added formatting changes and changes the description field as suggested.

Reply:
@pdcastro Pushed and resolving.